TLM/TLM/UI/TrafficManagerTool.cs

```diff
@@ -553,50 +770,53 @@ public struct NodeVisitItem {
-                nodeInput.m_ignoreNodeFlags = NetNode.Flags.None;
-                //nodeInput.m_ignoreNodeFlags = NetNode.Flags.Untouchable;
-
-                RaycastOutput nodeOutput;
-                if (RayCast(nodeInput, out nodeOutput)) {
+                var nodeInput = new RaycastInput(m_mouseRay, m_mouseRayLength) {
+                    m_netService = {
+                        // find road nodes
+                        m_itemLayers = ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels,
+                        m_service = ItemClass.Service.Road
+                    },
+                    m_ignoreTerrain = true,
+                    m_ignoreNodeFlags = NetNode.Flags.None
+                };
+
+                // nodeInput.m_netService2.m_itemLayers = ItemClass.Layer.Default
+                //     | ItemClass.Layer.PublicTransport | ItemClass.Layer.MetroTunnels;
+                // nodeInput.m_netService2.m_service = ItemClass.Service.PublicTransport;
+                // nodeInput.m_netService2.m_subService = ItemClass.SubService.PublicTransportTrain;
+                // nodeInput.m_ignoreNodeFlags = NetNode.Flags.Untouchable;
+
+                if (RayCast(nodeInput, out RaycastOutput nodeOutput)) {
                     HoveredNodeId = nodeOutput.m_netNode;
                 } else {
                     // find train nodes
-                    nodeInput.m_netService.m_itemLayers = ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
+                    nodeInput.m_netService.m_itemLayers =
+                        ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
                     nodeInput.m_netService.m_service = ItemClass.Service.PublicTransport;
                     nodeInput.m_netService.m_subService = ItemClass.SubService.PublicTransportTrain;
-                    nodeInput.m_ignoreTerrain = true;
                     nodeInput.m_ignoreNodeFlags = NetNode.Flags.None;
-                    //nodeInput.m_ignoreNodeFlags = NetNode.Flags.Untouchable;
+                    // nodeInput.m_ignoreNodeFlags = NetNode.Flags.Untouchable;
 
                     if (RayCast(nodeInput, out nodeOutput)) {
                         HoveredNodeId = nodeOutput.m_netNode;
                     } else {
                         // find metro nodes
-                        nodeInput.m_netService.m_itemLayers = ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
+                        nodeInput.m_netService.m_itemLayers =
+                            ItemClass.Layer.Default | ItemClass.Layer.MetroTunnels;
                         nodeInput.m_netService.m_service = ItemClass.Service.PublicTransport;
-                        nodeInput.m_netService.m_subService = ItemClass.SubService.PublicTransportMetro;
-                        nodeInput.m_ignoreTerrain = true;
+                        nodeInput.m_netService.m_subService =
+                            ItemClass.SubService.PublicTransportMetro;
```

Comment:
missing `nodeInput.m_ignoreTerrain`. Is `true` by default?

Reply:
It is set to true a few lines above. Rider IDE marks all repeated assignments to `true` as having no effect.